assets/javascripts/discourse/components/chat-toggle-membership-btn.js

```diff
@@ -0,0 +1,91 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import ChatApi from "discourse/plugins/discourse-chat/discourse/lib/chat-api";
+import { action, computed } from "@ember/object";
+
+export default class ChatToggleMembershipBtn extends Component {
+  @service chat;
+  tagName = "";
+
+  channel = null;
+  afterToggle = null;
+  isLoading = false;
+  labelType = null;
+  joinTitle = null;
+  joinIcon = null;
+  joinClasses = "";
+  leaveTitle = null;
+  leaveIcon = null;
+  leaveClasses = "";
+
+  @computed("channel.following")
```

Comment:
I wonder if we don't go a different way:

- have reasonable defaults for labels/icons
- allow you to override them when calling the component

```
<ToggleMembershipButton options=(hash leaveIcon="bell" joinLabel="foo") />
```

Reply:
Yeah that's what I was wondering whether an options hash would be better, will go with this!